src/api/routes/v1/get-network.ts

```diff
@@ -147,13 +161,20 @@ async function getNetworkFromUNL(unl: string): Promise<string | undefined> {
  * @returns The network associated with the public key of the node. Undefined
  * if the public key has not been seen by the network.
  */
-async function getNetworkFromPublicKey(
+async function getNetworkInfoFromPublicKey(
   publicKey: string,
-): Promise<string | undefined> {
+): Promise<NetworkInfo | undefined> {
   const result = await query('crawls')
     .select('public_key', 'networks')
     .where('public_key', '=', publicKey)
-  return result.length > 0 ? result[0].networks : undefined
+  const networkIds = result.length > 0 ? result[0].networks : undefined
+  if (networkIds == null) {
+    return networkIds
+  }
+  const networkResult = await query('networks')
+    .select('id', 'unls')
+    .whereIn('id', networkIds.split(','))
+  return networkResult
```

Comment:
perhaps, we should add a length check on the return of this query (or) return the `[0]`'th element. It appears that the comma-separated `networkIds` list could contain multiple elements.